Vj/hpomap

VarenyaJ opened the pull request.

pnrobinson committed:
q

pnrobinson committed:
:g
Mergnge branch 'master' of https://github.com/P2GX/prenatalppkt

pnrobinson committed:
hpo

VarenyaJ committed:
add all measurements from  into

VarenyaJ committed:
add comments about the structure of this file

VarenyaJ committed:
ruffing

SmartMonkey-git commented on src/prenatalppkt/phenotypic_export.py:
```diff
@@ -77,11 +77,11 @@ def __init__(
     # Mapping loader
     # ------------------------------------------------------------------ #
-    def _load_mappings(self, path: Path) -> dict:
+    def _load_mappings(self, path: Path) -> Dict:
```
This should stay dict. Its the up-to-date way of typing in python.

SmartMonkey-git commented on src/prenatalppkt/measurements/head_circumference_measurement.py:
```diff
@@ -8,18 +8,23 @@
+        super().__init__(measurement_type=BiometryType.HEAD_CIRCUMFERENCE, termbin_d=termbin_d)
 
     def name(self) -> str:
         """Return the canonical name for this measurement."""
```
This feels like a bad mix of aggregation and composition. The inheritance is not justified here, as there is not new functionality.

SmartMonkey-git reviewed (commented):
I have some suggestions. @VarenyaJ @pnrobinson 😄

VarenyaJ committed:
remove subclasses for measurements - we don't need them since they do not do anything special or unique from

VarenyaJ committed:
remove unnecessary files

VarenyaJ committed:
remove duplicate yaml

VarenyaJ committed:
- Add BiometryMappingLoader for YAML loading (Single Responsibility)
- Add PercentileRange and TermBin data structures
- Simplify SonographicMeasurement (no hardcoded if/elif)
- Update MeasurementEvaluation to use loader
- Simplify TermObservation to data-only class
- Update biometry_hpo_mappings.yaml with float-based ranges

VarenyaJ committed:
Add tests for new architecture

VarenyaJ committed:
remove old tests

VarenyaJ committed:
remove package version

VarenyaJ committed:
Introduce TermBin-backed HPO mapping and refactor percentile handling/exporter (modified biometry.py, mapping_loader.py, measurement_eval.py, measurements/term_bin.py, phenotypic_export.py, sonographic_measurement.py, term_observation.py and tests); TODO: fix mapping_loader doctest, finalize inclusive/exclusive percentile boundary policy, and update remaining phenotypic_export tests/fixtures to the new YAML schema

VarenyaJ committed:
rewrite yaml with floats of percentile ranges - transform a dict with tuple keys into a list of dicts

VarenyaJ committed:
update gitignore

VarenyaJ committed:
**Refactor percentile-to-HPO mapping architecture**

- Replace hardcoded percentile logic with data-driven `TermBin` and `PercentileRange` classes
- Introduce `BiometryMappingLoader` for YAML-based loading of percentile->HPO mappings
- Add `MeasurementEvaluation` factory for centralized initialization of measurement mappers
- Simplify `SonographicMeasurement` to be generic and configuration-driven (removed subclass hierarchy)
- Update `biometry_hpo_mappings.yaml` to float tuple range syntax (e.g. `(3,5)` instead of `between_3p_5p`)
- Refactor `phenotypic_export.py` to integrate new architecture
- Update and refactor corresponding tests (`test_phenotypic_export.py`, `test_term_bin.py`)
- Remove obsolete measurement subclasses and constants files
